src/azul/queues.py

```diff
@@ -235,10 +235,10 @@ def wait_to_stabilize(self) -> int:
             logger.info('Message count history (most recent first) is %r.',
                         list(reversed(total_lengths)))
 
-            if len(total_lengths) == total_lengths.maxlen:
+            if len(total_lengths) * sleep_time >= 60:
                 cummdiff = sum(
                     abs(first - second)
-                    for first, second in more_itertools.pairwise(total_lengths)
+                    for first, second in more_itertools.pairwise(list(total_lengths)[-6:])
```

Comment:
```suggestion
            total_lengths_full = len(total_lengths) == total_lengths.maxlen
            last_total_zero = len(total_lengths) and total_lengths[-1] == 0
            if total_lengths_full or last_total_zero:
                size = 6 if last_total_zero else len(total_lengths)
                cummdiff = sum(
                    abs(first - second)
                    for first, second in more_itertools.pairwise(list(total_lengths)[-size:])
```
Based on the ticket description I believe the desire is to keep the current behavior (compare all values in `total_lengths` _unless_ the queues are at `0`.
My suggestion could probably use some refactoring.

Reply:
Chiming to say:
1. I agree with @danielsotirhos 's interpretation of the ticket; loop should quit when (((10 minutes of no change) and (queues nonempty)) or ((1-2 minutes of no change) and (queues are empty)))
2. I think there's a typo on line 239. The call to `len` seems to be missing a comparison, otherwise it's redundant in evaluating the truthiness of `total_lengths`.

Reply:
`len(total_lengths)` is needed to ensure the -1 index won't throw a IndexError.
```
>>> total_lengths = []
>>> total_lengths[-1]
Traceback (most recent call last):
  File "<input>", line 1, in <module>
IndexError: list index out of range
```